### ISSUE TITLE
For discussion: Scoped storage sequence of updates

### ISSUE THREAD

**Issue by Arthur-Milchior** [ankidroid/Anki-Android#13146]
I previously thought we had the right to a single update where existing user could still accces ~/AnkiDroid. And that why everybody had to have migrated before we ever do a second update ever again.
Since it seems I'm wrong and we can let existing users postpone migration for as long as they want, I suspect that we could be able to have a journey of releases with smaller and simpler changes. Maybe I'm missing something, and if it's the case, I'd love to learn it.

# 1. Limit migrations to new installs

Assume #13145 is done; that is, users with new install can migrate from ~/AnkiDroid.
We remove the migration button, add a warning about uninstall, and we can release the app. Users who uninstall/reinstall will still be able to access their data. All other users see no difference. Then we can wait for bug report of bugs unrelated to migration. We'll certainly have some.

# 2. Add a migration for synced user.

Once the migration service works, we can, after a sync, add the option to migrate. This way, we should have plenty of users that are not technical users, but still, who have little chance of loosing their data since they are on ankiweb.

# 3. Add a back-up feature

For power user that don't want to sync with ankiweb, devise and implement a back-up feature that would allow them to potentially do a full back-up (media included).  Also offer to migrate once back-up is done.

# 4. Explain data safety

Warns users that they risk losing all data (a good idea even without scoped storage). We can decide later how much we want to force migration or whether we accept that they keep legacy storage

**Comment by jeffchang5** [comment]
[Migration Legacy Storage Docs](https://developer.android.com/training/data-storage/use-cases#migrate-legacy-storage)

It seems what Google wants to do is keep legacy storage for a couple of versions with migrations to Scoped Storage **for all users** . 

```After users install the new version of your app, they complete the data migration process on their devices. You can monitor the migration process across your user base by creating an analytics event.```

Past enough new releases, if enough users have migrated from legacy storage to scoped storage, remove the useLegacyStorage flag. A dialog or popup about this process for new updates makes sense to me to do some kind of backup.

**Comment by Arthur-Milchior** [comment]
Sorry @jeffchang5 , I missed your answer.

Thanks a lot for the explanation.

I believe any person who worked on this, e.g.,  who interacted with parts of https://github.com/orgs/ankidroid/projects/1 , already knew all of this. We have been working on it for more than a year. 
I was simply offering a path forward.
It was rejected on discord, so I close it.
